source/common/stats/thread_local_store.cc

```diff
@@ -61,6 +61,25 @@ std::list<GaugeSharedPtr> ThreadLocalStoreImpl::gauges() const {
   return ret;
 }
 
+std::list<HistogramSharedPtr> ThreadLocalStoreImpl::histograms() const {
+  // Handle de-dup due to overlapping scopes.
+  std::list<HistogramSharedPtr> ret;
+  std::unordered_set<std::string> names;
+  std::unique_lock<std::mutex> lock(lock_);
+  // TODO(ramaraochavali): incorporate the scopes into the histogram names.
+  for (ScopeImpl* scope : scopes_) {
+    for (auto histogram : scope->central_cache_.histograms_) {
+      const std::string& hist_name = histogram.first;
+      const ParentHistogramSharedPtr& parent_hist = histogram.second;
+      if (names.insert(hist_name).second) {
+        ret.push_back(parent_hist);
+      }
```

Comment:
else warn....I thinks if hist_name is duplicated something will be lost.

actually ASSERT would be even better.

Reply:
ASSERT is failing hot restart test cases. In hot restart case this might happen. So I removed the ASSERT. Any other ideas?